spec/controllers/api/items_controller_spec.rb

```diff
@@ -245,27 +246,229 @@
             expect(saved_items[3]).to match(items_param[3])
           end
         end
+
+        context 'and syncing items without the items_key_id field' do
+          it 'should set a default value for the field' do
+            @controller = Api::AuthController.new
+            post :sign_in, params: test_user_003_credentials
+
+            @controller = Api::ItemsController.new
+            request.headers['Authorization'] = "bearer #{JSON.parse(response.body)['token']}"
+
+            new_item = create(:item, :with_items_key_id, user_uuid: test_user.uuid)
+
+            items_param = [new_item].to_a.map(&:serializable_hash)
+            items_param[0].delete('items_key_id')
+
+            post :sync, params: { limit: 5, api: '20190520', items: items_param }
+
+            expect(response).to have_http_status(:ok)
+            expect(response.headers['Content-Type']).to eq('application/json; charset=utf-8')
+
+            parsed_response_body = JSON.parse(response.body)
+
+            expect(parsed_response_body).to_not be_nil
+
+            saved_items = parsed_response_body['saved_items']
+            expect(saved_items).to_not be_nil
+
+            saved_item = serialize_to_hash(saved_items)[0]
+            actual_item = Item.find(new_item[:uuid])
+
+            expect(saved_item[:items_key_id]).to be_nil
+            expect(actual_item[:items_key_id]).to be_nil
+          end
+        end
+
+        context 'and syncing items without the auth_hash field' do
+          it 'should set a default value for the field' do
+            @controller = Api::AuthController.new
+            post :sign_in, params: test_user_003_credentials
+
+            @controller = Api::ItemsController.new
+            request.headers['Authorization'] = "bearer #{JSON.parse(response.body)['token']}"
+
+            new_item = create(:item, :with_auth_hash, user_uuid: test_user.uuid)
```

Comment:
I'm not exactly sure about the details of the create function, but just to be sure, can we add an assertion here where we find the item and ensure that it does in fact have an auth_hash? i.e `expect(Item.find(uuid)[:auth_hash]).to_not be_nil`